Compilation error

When compiled a blank project with pyo3-polars 0.17 with feature dtype-categorical, a compilation error is generated:

error[E0432]: unresolved import polars_core::datatypes::create_enum_data_type
pyo3-polars-0.17.0/src/types.rs:6:5
6 | use polars_core::datatypes::create_enum_data_type;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^---------------------
  |     |                       |
  |     |                       help: a similar name exists in the module: create_enum_dtype
  |     no create_enum_data_type in datatypes

error[E0425]: cannot find function to_series in this scope
pyo3-polars-0.17.0/src/types.rs:465:30
    |
26  | pub(crate) fn get_series(obj: &Bound<'_, PyAny>) -> PyResult<Series> {
    | -------------------------------------------------------------------- similarly named function get_series defined here
...
465 |                 let series = to_series(py, PySeries(s));
    |                              ^^^^^^^^^ help: a function with a similar name exists: get_series

I've checked and in polars create_enum_data_type exists and it is enabled with dtype-categorical feature.
Same for get_series in pyo3-polars with same feature.

--- a/pyo3-polars/src/types.rs
+++ b/pyo3-polars/src/types.rs
@@ -2,8 +2,6 @@ use super::*;
 use crate::error::PyPolarsErr;
 use crate::ffi::to_py::to_py_array;
 use polars::export::arrow;
-#[cfg(feature = "dtype-categorical")]
-use polars_core::datatypes::create_enum_data_type;
 use polars_core::datatypes::{CompatLevel, DataType};
 use polars_core::prelude::*;
 use polars_core::utils::materialize_dyn_int;
@@ -592,7 +590,7 @@ impl<'py> FromPyObject<'py> for PyDataType {
                 let s = get_series(&categories.as_borrowed())?;
                 let ca = s.str().map_err(PyPolarsErr::from)?;
                 let categories = ca.downcast_iter().next().unwrap().clone();
-                create_enum_data_type(categories)
+                DataType::Enum(Some(RevMapping::build_local(categories)), Default::default())
             },
             "Date" => DataType::Date,
             "Time" => DataType::Time,

--- a/pyo3-polars/src/types.rs
+++ b/pyo3-polars/src/types.rs
@@ -341,7 +341,7 @@ impl IntoPy<PyObject> for PyExpr {
     }
 }
 
-#[cfg(feature = "dtype-full")]
+#[cfg(feature = "dtype-categorical")]
 pub(crate) fn to_series(py: Python, s: PySeries) -> PyObject {
     let series = SERIES.bind(py);
     let constructor = series
@@ -573,11 +573,12 @@ impl<'py> FromPyObject<'py> for PyDataType {
             #[cfg(feature = "dtype-categorical")]
             "Categorical" => {
                 let ordering = ob.getattr(intern!(py, "ordering")).unwrap();
-
-                let ordering = match ordering.extract::<&str>()? {
-                    "physical" => CategoricalOrdering::Physical,
-                    "lexical" => CategoricalOrdering::Lexical,
+                let ordering = ordering.extract::<PyBackedStr>()?;
+                let ordering = match  ordering.as_bytes() {
+                    b"physical" => CategoricalOrdering::Physical,
+                    b"lexical" => CategoricalOrdering::Lexical,
                     ordering => {
+                        let ordering = std::str::from_utf8(ordering).unwrap();
                         return Err(PyValueError::new_err(format!("invalid ordering argument: {ordering}")))
                     }
                 };
@@ -590,7 +591,7 @@ impl<'py> FromPyObject<'py> for PyDataType {
                 let s = get_series(&categories.as_borrowed())?;
                 let ca = s.str().map_err(PyPolarsErr::from)?;
                 let categories = ca.downcast_iter().next().unwrap().clone();
-                DataType::Enum(Some(RevMapping::build_local(categories)), Default::default())
+                DataType::Enum(Some(Arc::new(RevMapping::build_local(categories))), Default::default())
             },
             "Date" => DataType::Date,
             "Time" => DataType::Time,

